src/drive/web/modules/viewer/FilesViewer.jsx

```diff
@@ -49,15 +51,24 @@ class FilesViewer extends Component {
       })
   }
 
-  // If we get close of the last file fetched, but we know there are more in the folder
-  // (it shouldn't happen in /recent), we fetch 50 more files
-  fetchMoreIfNecessary() {
-    const { files, params, location, fetchMoreFiles, fileCount } = this.props
-    if (files.length === 0) return // we get there when loading a direct URL and the folder's content is still loading
+  // If we get close of the last file fetched (-5), but we know there are more in the folder
+  // (it shouldn't happen in /recent), we fetch FILES_FETCH_LIMIT more files
+  async fetchMoreIfNecessary() {
+    const { files, params, location, fetchMoreFiles } = this.props
     const currentIndex = files.findIndex(f => f.id === params.fileId)
-    if (files.length !== fileCount && files.length - currentIndex <= 5) {
-      const folderId = getFolderIdFromRoute(location, params)
-      fetchMoreFiles(folderId, files.length, FILES_FETCH_LIMIT)
+    const folderId = getFolderIdFromRoute(location, params)
+    // As fetchMoreIfNecessary is called every time, we had to check if we need to
```

Comment:
`every time` is not very clear here, Is it at every render, or every mount, or something else?